src/login/logind.c

```diff
@@ -771,7 +772,14 @@ static int manager_vt_switch(sd_event_source *src, const struct signalfd_siginfo
 
         active = m->seat0->active;
         if (!active || active->vtnr < 1) {
-                log_warning("Received VT_PROCESS signal without a registered session on that VT.");
+                int r;
+
+                log_warning("Received VT_PROCESS signal without a registered session, restoring VT.");
+
+                r = restore_terminal("/dev/tty0", true);
```

Comment:
why /dev/tty0? why not the tty of `active` directly?

Reply:
I thought that both were equivalent at that point.

Reply:
In fact `active` has no tty at this point, does it ?

Reply:
well, the user might have switched to the next VT by this time, hence we should use the right VT, and not rely on the kernel mapping this for us.

Reply:
> the user might have switched to the next VT by this time

It's very unlikely as someone needs to acknowledge VT_PROCESS signal sent by the kernel in order to make the VT switch work again. And logind is supposed to do that.

> we should use the right VT, and not rely on the kernel mapping this for us.

OK but how can we do that exactly since at this point `active->vtnr < 1` so I suppose `active->vtfd` is no more relevant ?

Reply:
This definitely deserves an explanatory comment in the sources.

Reply:
OK